tests/Neo.UnitTests/Cryptography/UT_Crypto.cs

```diff
@@ -94,6 +97,133 @@ public void TestSecp256k1()
+            var expectedPubKey2 = "02dbf1f4092deb3cfd4246b2011f7b24840bc5dbedae02f28471ce5b3bfbf06e71".HexToBytes();
+
+            var recoveredKey2 = Crypto.ECRecover(signature2, messageHash2);
+            recoveredKey2.EncodePoint(true).Should().BeEquivalentTo(expectedPubKey2);
+
+            // Test case 3 - recovery param 0
+            var message3 = "db0d31717b04802adbbae1997487da8773440923c09b869e12a57c36dda34af11b8897f266cd81c02a762c6b74ea6aaf45aaa3c52867eb8f270f5092a36b498f88b65b2ebda24afe675da6f25379d1e194d093e7a2f66e450568dbdffebff97c4597a00c96a5be9ba26deefcca8761c1354429622c8db269d6a0ec0cc7a8585c".HexToBytes();
+            var messageHash3 = "c36d0ecf4bfd178835c97aae7585f6a87de7dfa23cc927944f99a8d60feff68b".HexToBytes();
+            var signature3 = "f25b86e1d8a11d72475b3ed273b0781c7d7f6f9e1dae0dd5d3ee9b84f3fab89163d9c4e1391de077244583e9a6e3d8e8e1f236a3bf5963735353b93b1a3ba93500".HexToBytes();
+            var expectedPubKey3 = "03414549fd05bfb7803ae507ff86b99becd36f8d66037a7f5ba612792841d42eb9".HexToBytes();
+
+            var recoveredKey3 = Crypto.ECRecover(signature3, messageHash3);
+            recoveredKey3.EncodePoint(true).Should().BeEquivalentTo(expectedPubKey3);
+
+            // Test invalid cases
+            var invalidSignature = new byte[65];
+            Action action = () => Crypto.ECRecover(invalidSignature, messageHash1);
+            action.Should().Throw<ArgumentException>();
+
+            // Test with invalid recovery value
+            var invalidRecoverySignature = signature1.ToArray();
+            invalidRecoverySignature[64] = 29; // Invalid recovery value
+            action = () => Crypto.ECRecover(invalidRecoverySignature, messageHash1);
+            action.Should().Throw<ArgumentException>();
+
+            // Test with wrong message hash
+            var recoveredWrongHash = Crypto.ECRecover(signature1, messageHash2);
+            recoveredWrongHash.EncodePoint(true).Should().NotBeEquivalentTo(expectedPubKey1);
+        }
+
+        [TestMethod]
+        public void TestERC2098()
+        {
+            //
+            // Private Key: 0x1234567890123456789012345678901234567890123456789012345678901234
+            // Message: "It's a small(er) world"
+            // Signature:
+            // r:  0x9328da16089fcba9bececa81663203989f2df5fe1faa6291a45381c81bd17f76
+            // s:  0x139c6d6b623b42da56557e5e734a43dc83345ddfadec52cbe24d0cc64f550793
+            // v:  28
+
+
+            // Test from https://eips.ethereum.org/EIPS/eip-2098
+            var privateKey = "1234567890123456789012345678901234567890123456789012345678901234".HexToBytes();
+
+            var expectedPubKey1 = (Neo.Cryptography.ECC.ECCurve.Secp256k1.G * privateKey).ToArray();
+
+            Console.WriteLine($"Expected PubKey: {expectedPubKey1.ToHexString()}");
+            var message1 = Encoding.UTF8.GetBytes("Hello World");
+            var messageHash1 = message1.Keccak256();
```

Comment:
This is supposed to be an Ethereum signed message which is not explicitly stated in ERC-2098
```
function toEthSignedMessageHash(bytes memory message) internal pure returns (bytes32) {
        return
            keccak256(bytes.concat("\x19Ethereum Signed Message:\n", bytes(Strings.toString(message.length)), message));
}
```
Expected value: `0xa1de988600a42c4b4ab089b619297c17d53cffae5d5120d82d8a92d0bb3b78f2`

Reply:
Seems we need to simulate solidity behavior and insert billions of 0x00